### ISSUE TITLE
"Contributors Wanted" badge does not provide explanation

### ISSUE THREAD

**Issue by vossad01** [scalacenter/scaladex#309]
Ideally, the "Contributors Wanted" badge should be hyperlinked to a project-specific page that directs a user how to contribute.

This may be:

- The Project's Issue tracker
- The Project's CONTRIBUTING.md file
- Some other project-specific resource that gives direction on how to get involved.

In the absence of such direction the badge does not really provide much information and may even be hurtful to projects that don't have it.  Perhaps this view is not shared, but when a project is on GitHub, I assume that it wants contributors unless it indicates otherwise.  The badge on Scaladex suggests the default is the other way around: projects don't want contributors unless they say they do.

**Comment by heathermiller** [comment]
I agree that we should give the library author an option of something to link to when someone clicks on the "contributors wanted" badge. Makes sense 👍 

Though to respond to this point:

> In the absence of such direction the badge does not really provide much information and may even be hurtful to projects that don't have it. Perhaps this view is not shared, but when a project is on GitHub, I assume that it wants contributors unless it indicates otherwise.

Nah, actually this isn't true. There's actually numbers and research done on this (I'd have to dig around for a while to find citations for you), but it's actually very common for companies or individuals to post code on GitHub for the sake of open sourcing it (for others to fork), but with little or no intentions of accepting contributions. In fact, many companies even have internal policies not to accept outside contributions (this is the case for many Microsoft projects that have been open sourced, for example). Yet, on the other hand, there are libraries that are scrambling for help, and there's no way to visually indicate on which side of the spectrum some library is. 

So I would disagree that the "contributors wanted" badge is bad thing. Instead, I think we could invest more effort here helping library authors to communicate their needs to potential contributors via Scaladex... Help wanted brainstorming what we could do to better help this! 😄

**Comment by MasseGuillaume** [comment]
we could link to the following search: 
https://index.scala-lang.org/search?q=contributorsWanted%3Atrue

**Comment by vossad01** [comment]
@MasseGuillaume I think that is best for now as it assigns some more meaning to the badge.

My best ideas on how to improve the help wanted badge towards making it useful are in the OP.

[Up For Grabs [.net]](http://up-for-grabs.net/#/) is the dominant player I know in this space.  It establishes a convention and requires a commitment in grooming issues.  I have looked in the past at submitting more Scala projects to it, but for the handful I looked at, I was not convinced of my ability to correctly identify a tag meeting the appropriate requirements.

[First Timers Only](http://www.firsttimersonly.com/) is worth mentioning in this context as a related initiative though somewhat tangential.

Additional text in README.md concerning contributing is always a good sign.  CONTRIBUTING.md goes a long ways in setting expectations, and does get visual attention on GitHub, but not until a user takes an action towards contributing (Issue/PR).

@heathermiller This is an aside, but if you do come across them I would be interested in those.  I wonder if consider the selection bias that occurs occurs in actual use.  That is, does it remain true for the subset of GitHub projects that people might encounter and want to contribute to.  There are a lot of reason that code (and non-code) gets "dumped" on GitHub and a random sampling would find many that would not want contributions.  No one would really expect a pull request on someone's work following a tutorial, for example.   I did just a quick Google Scholar just for "GitHub" in case it was obvious.  Relevantly, ["The promises and perils of mining GitHub"](http://dl.acm.org/citation.cfm?id=2597074) came up.

Especially historically, Microsoft has released source for many things with out the intention they be developed or maintained in the community, Enterprise Library, would be a good example.  They probably still do that with some things, but there are many popular instances where this is not the case such as TypeScript, Visual Studio Code, Roslyn (compiler), and .NET all encouraging contribution.

**Comment by heathermiller** [comment]
I'd suggest we just add a radio select to the edit page, if "contributors wanted" is selected. We could let people select "CONTRIBUTING.md" or "Other" and if "Other" is selected, we could have a field containing where to link to.

WDYT @MasseGuillaume and @vossad01?

**Comment by heathermiller** [comment]
Ah, good idea @vossad01 to do something similar to _Up for Grabs_ or _First Timers Only_. We could just query a site for tickets with these tags and have another index page indexing the projects with such projects. 

Cool idea! Maybe we should rename this ticket.

**Comment by vossad01** [comment]
@heathermiller That sounds like a good plan, but I don't fully understand your vision.  Do you think you could create a new issue for it?

I think getting more Scala projects listed on _Up For Grabs_ would be beneficial and Scaladex may be a good way to go about making that happen and showcase Scala projects with relevant issues.

What I am not sure about is if you are proposing removing the current Contributors Wanted badge to something that depends on the project adopting a tagging scheme, or if it is another complementary feature.

If it is a replacement, we can let this one be resolved by #312 and the new issue will carry on.  If it is complementary, then I think [the behavior from your post](https://github.com/scalacenter/scaladex/issues/309#issuecomment-269676545) would still be beneficial and this issue can track that.

**Comment by MasseGuillaume** [comment]
I think we had a similar idea in the past: https://github.com/scalacenter/scaladex/issues/112
